queue.h

```diff
@@ -0,0 +1,48 @@
+#ifndef CONCURRENT_QUEUE_H
+#define CONCURRENT_QUEUE_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifdef _WIN32
+#if defined(CQ_STATIC) //preferred way
+#define CQ_EXPORT
+#elif defined(DLL_EXPORT)
+#define CQ_EXPORT __declspec(dllexport)
+#else
+#define CQ_EXPORT __declspec(dllimport)
+#endif
+#else
+/*
+need non-windows version here
+*/
+#endif
+
+#define CQ_OK 0
+#define CQ_ERR -1
+#define CQ_BAD_HANDLE -2
+#define CQ_BAD_VALUE_HANDLE -3
+#define CQ_VALUE_NOT_FOUND 1
+
+typedef void* CQHandle;
+typedef void* BCQHandle;
+typedef void* CQValue;
+
+CQ_EXPORT int cq_create(CQHandle* handle);
+CQ_EXPORT int cq_destroy(CQHandle* handle);
+CQ_EXPORT int cq_enqueue(CQHandle* handle, CQValue* value);
```

Comment:
Why is `value` a pointer to a pointer?

Reply:
no good reason.

Reply:
it adds an extra level of indirection, which is both slower and more cumbersome

Reply:
agreed, changing.